pkg/controller.v1/control/podgroup_control.go

```diff
@@ -193,3 +193,29 @@ func (s *SchedulerPluginsControl) CreatePodGroup(podGroup client.Object) error {
 }
 
 var _ PodGroupControlInterface = &SchedulerPluginsControl{}
+
+// KoordinatorControl is the  implementation of PodGroupControlInterface with scheduler-plugins.
+type KoordinatorControl struct {
+	*SchedulerPluginsControl
+}
+
+func (s *KoordinatorControl) DecoratePodTemplateSpec(pts *corev1.PodTemplateSpec, job metav1.Object, _ string) {
+	if len(pts.Spec.SchedulerName) == 0 {
+		pts.Spec.SchedulerName = s.GetSchedulerName()
+	}
+	if pts.Labels == nil {
+		pts.Labels = make(map[string]string)
+	}
+	pts.Labels[schedulerpluginsv1alpha1.PodGroupLabel] = job.GetName()
+}
```

Comment:
> why is the PR bigger than just this line?
> 
> if we can have this, I would support this change. Otherwise, I'm opposed.

@alculquicondor We need to add this function if we natively support the koordinator-scheduler.
So, you may oppose this change.

Reply:
This is no different from SchedulerPluginsControl. They can remove `KoordinatorControl` and use the existing implementation.

Reply:
Sorry for your confusion.
We must handle another value in the function GetSchedulerName (L212 - L214 in this file).

This means we must specify the `koord-scheduler` to the schedulerName in podSpec.

Reply:
if you treat `SchedulerPluginsControl` as the default implementation when the gang-scheduler-plugin configuration is set, you can pass that.

Reply:
> if you treat `SchedulerPluginsControl` as the default implementation when the gang-scheduler-plugin configuration is set, you can pass that string, even if it's not "scheduler-plugins"

I think this is ok,add --enable-gang-scheduler and --gang-scheduler-name. SchedulerPluginsControl as the default implementation when the enable-gang-scheduler is true, and set gang-scheduler-name if it is not nil.

Reply:
just not empty gang-scheduler-name should be enough. No need for an extra enable-gang-scheduler

Reply:
sgtm

Reply:
> scheduler-plugins as secondary scheduler, set gang-scheduler-name=scheduler-plugins(or any other name they use. like koordinator.) scheduler-plugins as default scheduler, set gang-scheduler-name=default-scheduler volcano as secondary scheduler, set gang-scheduler-name=volcano @tenzen-y @alculquicondor WDYT

@Syulin7 I have a question. If we set gang-scheduler-name=default-scheduler, does the training-operator embed `default-scheduler` in podTemplateSpec.spec.schedulerName? Or does the training-operator skip set a value in podTemplateSpec.spec.schedulerName?

Reply:
@tenzen-y  I think if we set gang-scheduler-name=default-scheduler, the training-operator should skip set a value in podTemplateSpec.spec.schedulerName.

Reply:
sgtm